Source/ACE/Entity/Generator.cs

```diff
@@ -9,7 +9,8 @@ namespace ACE.Entity
     public class Generator : WorldObject
     {
         public Generator(ObjectGuid guid, AceObject baseAceObject)
-            : base((ObjectType)baseAceObject.Type, guid)
+            // : base((ObjectType)baseAceObject.Type, guid)
+            : base(guid, baseAceObject)
```

Comment:
Given your new constructor, many of the data copying below (e.g. your Animation Texture, and Pallette overrides) are now redundant.